File: renumber.py
import glob
import os
import sys

def get_target_list():
    files = glob.glob('../../screencapture/*.png')

    target_list = []
    for file in files:
        filename = os.path.basename(file)
        parts = filename.split('_')
        target_list.append(parts[0])
        #print file

    print list(set(target_list))

def renumber(target_dir, target_date):
    #target_date = '20140217'
    #target_dir = '../../screencapture' + '/'
    target_dir = target_dir + '/'

    target_list = []
    #print('%s%s_00000.png' % (target_dir, target_date))
    for file in glob.glob('%s%s_*.png' % (target_dir, target_date)):
        filename = os.path.basename(file)
        target_list.append(filename)
        #print file

    target_list.sort()
    number = 1
    for filename in target_list:
        old = filename
        new = '%s_%05d.png' % (target_date, number)
        print('%s -> %s' % (old, new))
        if old != new:
            os.rename(target_dir + old, target_dir + new)
        number = number + 1

#get_target_list()
renumber(sys.argv[1], sys.argv[2])

File: renumber.go
package main

import (
	"fmt"
	"os"
	"path/filepath"
	"sort"
)

func renumber(target_dir string, target_date string) {
	files, err := filepath.Glob(target_dir + target_date + "_*.png")
	if err != nil {
		panic(err)
	}

	target_list := []string{}
	for _, file := range files {
		basename := filepath.Base(file)
		target_list = append(target_list, basename)
	}

	sort.Strings(target_list)

	finished := make(chan bool)

	chan_count := 0
	for n, bf := range target_list {
		num := n
		before_filename := bf
		after_filename := fmt.Sprintf("%s__%05d.png", target_date, num+1)

		// 並列化のジレンマについて
		// 並列実行すると、処理順序の関係でファイルを上書きしてしまう可能性があるため
		// 書き換え先のファイル名はかぶらないようにする必要がある。
		// その場合、すべてのファイルをrenameするため、ずれてなければpassできた直列実行よりも遅くなる
		//if before_filename != after_filename {
		go func() {
			fmt.Println(before_filename + " -> " + after_filename)
			os.Rename(target_dir+before_filename, target_dir+after_filename)
			finished <- true
		}()
		chan_count += 1
		//}
	}

	// 終わるまで待つ
	for i := 1; i <= chan_count; i++ {
		<-finished
	}
}

func main() {
	renumber(os.Args[1], os.Args[2])
}
